dspy/clients/lm.py

```diff
@@ -60,9 +87,84 @@ def __call__(self, prompt=None, messages=None, **kwargs):
         
         return outputs
 
+    def copy(self, **kwargs):
+        """Create a copy of the current LM with updated kwargs."""
+        kwargs = {
+            "model": self.model,
+            "model_type": self.model_type,
+            "cache": self.cache,
+            "launch_kwargs": self.launch_kwargs,
+            **self.kwargs,
+            **kwargs,
+        }
+        return self.__class__(**kwargs)
+    
     def inspect_history(self, n: int = 1):
         _inspect_history(self, n)
 
+    def launch(self):
+        """Send a request to the provider to launch the model, if needed."""
+        if is_self_hosted_model(self.model):
+            self_hosted_model_launch(self.model, self.launch_kwargs)
+        elif is_anyscale_model(self.model):
+            anyscale_model_launch(self.model, self.launch_kwargs)
+        msg = f"`launch()` is called for the auto-launched model {self.model}"
+        msg += " -- no action is taken!"
+        logger.info(msg)
+
+    def kill(self):
+        """Send a request to the provider to kill the model, if needed."""
+        if is_self_hosted_model(self.model):
+            self_hosted_model_kill(self.model, self.launch_kwargs)
+        elif is_anyscale_model(self.model):
+            anyscale_model_kill(self.model, self.launch_kwargs)
+        msg = f"`kill()` is called for the auto-launched model {self.model}"
+        msg += " -- no action is taken!"
+        logger.info(msg)
+
+    def finetune(self,
+            train_method: TrainingMethod,
+            train_data: List[Dict[str, Any]],
+            train_kwargs: Optional[Dict[str, Any]]=None,
+            cache_finetune: bool = True,
+            provider: str = "openai",
+        ) -> FinetuneJob:
+        """Start model fine-tuning, if supported."""
+        # Fine-tuning is experimental and requires the experimental flag
+        from dspy import settings as settings
+        err = "Fine-tuning is an experimental feature."
+        err += " Set `dspy.settings.experimental` to `True` to use it."
+        assert settings.experimental, err
+
+        # Initialize the finetune job
+        FinetuneJobClass = get_provider_finetune_job_class(provider=provider)
+        finetune_job = FinetuneJobClass(
```

Comment:
Should we also pass `train_method` here? nit. Let's pick between `method` vs. `train_method` and change all the functions to use the same name

Reply:
Lets use train method